pymc3/sampling.py

```diff
@@ -265,20 +265,26 @@ def _choose_backend(trace, chain, shortcuts=None, **kwds):
     except KeyError:
         raise ValueError('Argument `trace` is invalid.')
 
+def _make_parallel(arg, njobs):
+    if not shape(arg):
```

Comment:
This really leads to odd and hard to debug problems down the lines. For example, pandas changed the __bool__ eval of dataframes and it's not guaranteed that that won't happen in the future. It might be OK here so if you think it's fine you can go ahead.